src/Game/Main.elm

```diff
@@ -131,19 +119,98 @@ updatePlayer key player =
             player
 
 
-updateOrbs : Player -> List ( Int, Int ) -> List ( Int, Int )
+updateOrbs : Player -> List Point -> List Point
 updateOrbs player orbs =
     let
+        orbAvoidPlayer : Point -> Point
+        orbAvoidPlayer orb =
+            if orb == move player.direction player.position then
+                orb
+
+            else
+                orb
+                    |> move (invertDirection player.direction)
+
+        orbAvoidOrbs allOrbs orb =
+            if List.any (\otherOrb -> otherOrb == orb) allOrbs then
+                orb
+                    |> move (invertDirection player.direction)
+                    |> move (invertDirection player.direction)
+
+            else
+                orb
+
         f orb =
             if player.position == orb then
                 move player.direction orb
+                    |> constrainToWorld
+                    |> orbAvoidPlayer
+                    |> orbAvoidOrbs orbs
 
             else
                 orb
     in
     List.map f orbs
```

Comment:
The only thing I've changed here is to remove the `allOrbs` argument to the `orbAvoidOrbs` function.
It's not needed since we already have access to `orbs` in this scope.

```suggestion
updateOrbs : Player -> List Point -> List Point
updateOrbs player orbs =
    let
        orbAvoidPlayer : Point -> Point
        orbAvoidPlayer orb =
            if orb == move player.direction player.position then
                orb

            else
                orb
                    |> move (invertDirection player.direction)

        orbAvoidOrbs : Point -> Point
        orbAvoidOrbs orb =
            if List.any (\otherOrb -> otherOrb == orb) orbs then
                orb
                    |> move (invertDirection player.direction)
                    |> move (invertDirection player.direction)

            else
                orb

        f orb =
            if player.position == orb then
                move player.direction orb
                    |> constrainToWorld
                    |> orbAvoidPlayer
                    |> orbAvoidOrbs

            else
                orb
    in
    List.map f orbs

```

Reply:
Also, seems like we can remove the `randomNumbers` field in the `Model`

Reply:
That's great! Feel free to be even more nitpicky, it's great for learning. Funny thing I also have done some of the same changes, so now I will try to merge these changes together. This is my first time using pull requests, which I'm quite excited about!